pkg/registry/core/shoot/storage/kubeconfig.go

```diff
@@ -106,8 +106,16 @@ func (r *KubeconfigREST) Create(ctx context.Context, name string, obj runtime.Ob
 		return nil, apierrors.NewInternalError(fmt.Errorf("cannot convert to *core.Shoot object - got type %T", shootObj))
 	}
 
-	if len(shoot.Status.AdvertisedAddresses) == 0 {
-		fieldErr := field.Invalid(field.NewPath("status", "status"), shoot.Status.AdvertisedAddresses, "no kube-apiserver advertised addresses in Shoot .status.advertisedAddresses")
+	// filter only addresses that actually advertise the kube-apiserver
+	// it is possible that the list of addresses also include URLs like the shoot's issuer URL
+	kapiAddresses := []core.ShootAdvertisedAddress{}
+	for _, addr := range shoot.Status.AdvertisedAddresses {
+		if addr.Name == "external" || addr.Name == "internal" || addr.Name == "unmanaged" {
```

Comment:
Perhaps there should be constants for these names so that they can be reused here and where they are written?

Reply:
Done with https://github.com/gardener/gardener/pull/9196/commits/6fb40b706a27f038bd7cdf97a2c5a7d75a87679c